collect_app/src/main/java/org/odk/collect/android/dao/InstancesDao.java

```diff
@@ -145,6 +146,41 @@ public void deleteInstancesDatabase() {
         Collect.getInstance().getContentResolver().delete(InstanceProviderAPI.InstanceColumns.CONTENT_URI, null, null);
     }
 
+    public void deleteInstancesFromIDs(String ids[]){
+        int count = ids.length;
+        int i = 0 ;
```

Comment:
Yes it can be done, how i missed this i saw same thing was done while doing slicing in upload also in processChunk method. Thanks for pointing out @nribeka